### PR TITLE
Multiuser Console

### PR DESCRIPTION
**Pull request by mofirouz**

### CODE REVIEW

File: migrate/sql/20201005180855-console.sql [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * limitations under the License.
+ */
+
+-- +migrate Up
+CREATE TABLE IF NOT EXISTS console_users (
+ PRIMARY KEY (id),
+
+ id           UUID          NOT NULL,
+ username     VARCHAR(128)  CONSTRAINT users_username_uniq UNIQUE NOT NULL,
+ email        VARCHAR(255)  CONSTRAINT users_email_uniq UNIQUE NOT NULL,
```
Comment (sesposito) [inline migrate/sql/20201005180855-console.sql:23]: Are we going to use email for console users accounts?

File: migrate/sql/20201005180855-console.sql [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ password     BYTEA         CHECK (length(password) < 32000),
+ metadata     JSONB         DEFAULT json_build_object() NOT NULL,
+ create_time  TIMESTAMPTZ   DEFAULT now() NOT NULL,
+ update_time  TIMESTAMPTZ   DEFAULT now() NOT NULL,
+ disable_time TIMESTAMPTZ   DEFAULT '1970-01-01 00:00:00 UTC' NOT NULL
+);
```
Comment (sesposito) [inline migrate/sql/20201005180855-console.sql:29]: I think we are missing the `role` column.

File: server/console_account.go [hunk 3]
```diff
@@ -208,6 +256,188 @@ func (s *ConsoleServer) GetWalletLedger(ctx context.Context, in *console.Account
...
+					{
+						Id:         in.Filter,
+						UpdateTime: &timestamp.Timestamp{Seconds: createTime.Time.Unix()},
+					},
+				},
+				TotalCount: countAccounts(ctx, s.logger, s.db),
+			}, nil
+		}
+
+		query := "SELECT user_id, create_time FROM user_tombstone LIMIT 50"
```
Comment (sesposito) [inline server/console_account.go:299]: Will this need to have pagination or we'll just return the first 50 results?

File: server/console_account.go [hunk 4]
```diff
@@ -208,6 +256,188 @@ func (s *ConsoleServer) GetWalletLedger(ctx context.Context, in *console.Account
...
+			query += " AND disable_time <> '1970-01-01 00:00:00 UTC'"
+		}
+
+		rows, err := s.db.QueryContext(ctx, query, params...)
+		if err != nil {
+			s.logger.Error("Error querying users.", zap.Any("in", in), zap.Error(err))
+			return nil, status.Error(codes.Internal, "An error occurred while trying to list users.")
+		}
+
+		users := make([]*api.User, 0, 2)
```
Comment (sesposito) [inline server/console_account.go:353]: why allocate a capacity for 2 users, shouldn't the filter only return one?

**Review by sesposito — COMMENTED** [review]
I've noticed we aren't doing access control on the account Ban/Unban/Delete APIs implemented in the PR, is it to be added later?

File: migrate/sql/20201005180855-console.sql (hunk 1 above)
Comment (mofirouz) [reply migrate/sql/20201005180855-console.sql:23]: Mixture of both.

File: migrate/sql/20201005180855-console.sql (hunk 2 above)
Comment (mofirouz) [reply migrate/sql/20201005180855-console.sql:29]: Good catch.

File: server/console_account.go (hunk 3 above)
Comment (mofirouz) [reply server/console_account.go:299]: Will need pagination. I didn't actually edit any of the Go code.

File: server/console_account.go (hunk 4 above)
Comment (mofirouz) [reply server/console_account.go:353]: The code requires revising when it is implemented properly.

**Review by sesposito — APPROVED** [review]

**Pull request closed without merging**